app/Models/User.php

```diff
@@ -539,7 +576,9 @@ public function removeFromGroup(Group $group, ?self $actor = null): void
 
     public function setDefaultGroup(Group $group, ?self $actor = null): void
     {
-        $this->addToGroup($group, $actor);
+        if (!$this->isGroup($group)) {
```

Comment:
forgot about this but `isGroup` shouldn't be used here as it's only for authorization check (value depends on auth method), not actual state of the user.